src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -588,7 +588,7 @@ public override async ValueTask<(TValue value, string etag)> GetStateAndETagAsyn
                 await this.MakeSaveStateCallAsync(storeName, key, value, etag, stateOptions, metadata, cancellationToken);
                 return true;
             }
-            catch (RpcException)
+            catch (RpcException rpc) when (rpc.Status.Detail.Contains(ErrStateSave))
```

Comment:
We need to find a better way to do this. These strings are intended for human consumption and will change in the future. These should be programmatic error codes not human readable text.

Reply:
I agree but I guess that would need changes in the runtime which might affect all sdks.

Reply:
Let's start a discussion about it then. I don't think this is a viable approach.

Reply:
If we only want `TrySaveStateAsync` to return false if there's an OCC conflict (which I think is the intended behavior), then the `ErrStateSave` string won't even be precise enough to determine what happened. Any error returning from a state component while saving keys will be surfaced as an ERR_STATE_SAVE message.
I think the only way to really solve this is to implement https://github.com/dapr/dapr/issues/2619

Reply:
I tested this with redis and Dapr runtime seems to be getting the error below from redis component. I checked the component code and this seems to be what the redis is returning:-
== DAPR == @@@@@@ state err: failed to set key DaprClient||mykey: ERR Error running script (call to f_83e03ec05d6a3b6fb48483accf5e594597b6058f): @user_script:1: user_script:1: failed to set key DaprClient||mykey

Not very useful :(

Reply:
yeah I dug into the code in components contrib, and the runtime doesn't make a distinction. Any error returned by the state store is treated the same. Let's discuss this using dapr/dapr#2619

Reply:
https://github.com/dapr/dapr/pull/2647/files#diff-ce3f68305f0eb28b6b8911c281186536c80d8cac9dff3cece2c8bdf3dd955315R499